sycl/test/check_device_code/vector/vector_math_ops.cpp

```diff
@@ -91,6 +129,29 @@ SYCL_EXTERNAL auto TestAdd(vec<std::byte, 8> a, vec<std::byte, 8> b) {
 // CHECK:       _ZN4sycl3_V1plERKNS0_3vecIbLi4EEES4_.exit:
 // CHECK-NEXT:    store <4 x i8> [[VECINS_I_I6_I_I]], ptr addrspace(4) [[AGG_RESULT]], align 4, {{.*}}
 // CHECK-NEXT:    ret void
+// CHECK-ARR-STORAGE-LABEL: define dso_local spir_func void @_Z7TestAddN4sycl3_V13vecIbLi4EEES2_(
+// CHECK-ARR-STORAGE-SAME: ptr addrspace(4) dead_on_unwind noalias nocapture writable sret(%"class.sycl::_V1::vec.6") align 4 [[AGG_RESULT:%.*]], ptr nocapture noundef readonly byval(%"class.sycl::_V1::vec.6") align 4 [[A:%.*]], ptr nocapture noundef readonly byval(%"class.sycl::_V1::vec.6") align 4 [[B:%.*]]) {{.*}} {
+// CHECK-ARR-STORAGE-NEXT:  entry:
+// CHECK-ARR-STORAGE-NEXT:    tail call void @llvm.experimental.noalias.{{.*}}
```

Comment:
`ext_vector_type` for bool is literally a dense sequence of bits. That's why we have a for loop after `[[ADD_I:%.*]] = add <4 x i8> [[TMP0]], [[TMP1]]` to sign extend.
I'm wondering if, for booleans, we should just perform the operation on std::array<bool> directly?